From hjl@lucon.org  Mon Mar 17 18:17:08 1997
Return-Path: <hjl@lucon.org>
Received: from elvis.sw.ods.com ([160.86.13.7]) by elo.sw.ods.com
	 with esmtp id m0w6maS-00013tC
	(Debian Smail-3.2 1996-Jul-4 #2); Mon, 17 Mar 1997 18:17:08 -0600 (CST)
Received: from ocean.lucon.org ([204.160.189.73]) by elvis.sw.ods.com
          (Netscape Mail Server v2.01) with SMTP id AAA12906
          for <david@sw.ods.com>; Mon, 17 Mar 1997 18:18:19 -0500
Received: by ocean.lucon.org
	id m0w6ma7-0004GqC
	(Debian /\oo/\ Smail3.1.29.1 #29.37); Mon, 17 Mar 97 16:16 PST
Message-Id: <m0w6ma7-0004GqC@ocean.lucon.org>
From: hjl@lucon.org (H.J. Lu)
Subject: Re: Bug#8126: New NIS *almost* works
To: david@sw.ods.com (David Engel)
Date: Mon, 17 Mar 1997 16:16:47 -0800 (PST)
Cc: bofh@bofh.me.umn.edu
In-Reply-To: <19970317165115.03314@sw.ods.com> from "David Engel" at Mar 17, 97 04:51:15 pm
X-Mailer: ELM [version 2.4 PL25 PGP2]
MIME-Version: 1.0
Content-Type: text/plain; charset=US-ASCII
Content-Transfer-Encoding: 7bit
Status: RO
Content-Length: 1191
Lines: 46

> 
> H.J.,
> 
> Here's a bug report I got on the new ypbind.
> 
> David
> 
> Package: nis
> Version: 2.10-1.1
> 
> Well, I downloaded and tested the new nis package, and neither 'ypset'
> nor 'ybind -S domainname,server' work (both core dump).
> 
> ypbind still works fine if there is an NIS server on the same network
> (so it can find it by broadcast).
> 
> (Unfortunately, this means two of my systems I have been unable to get
> working).
> 

Please try this patch against my ypclients and let me know the
result. BTW, I cannot reproduce the core dump with
"ybind -S domainname,server". Please let me know how to
duplicate the bug.

Thanks.


H.J.
---
Index: ypset/ypset.c
===================================================================
RCS file: /home/work/cvs/nis/ypclients/ypset/ypset.c,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 ypset.c
--- ypset.c	1997/01/24 00:20:21	1.1.1.1
+++ ypset.c	1997/03/17 23:47:05
@@ -95,7 +95,7 @@
   else
     bcopy (&server_addr, &ypsd.ypsetdom_addr, sizeof (server_addr));
   
-  strncpy(ypsd.ypsetdom_domain, dom, sizeof ypsd.ypsetdom_domain);
+  ypsd.ypsetdom_domain = dom;
   ypsd.ypsetdom_port = port;
   ypsd.ypsetdom_vers = YPVERS;
   

